src/sentry/workflow_engine/migration_helpers/alert_rule.py

```diff
@@ -575,64 +606,85 @@ def dual_update_migrated_alert_rule(alert_rule: AlertRule, updated_fields: dict[
-            # this shouldn't be possible due to the way we dual write
-            logger.error(
-                "AlertRuleDetector has no associated DataConditionGroup",
-                extra={"alert_rule_id": alert_rule.id},
-            )
-            raise MissingDataConditionGroup
-        data_conditions = DataCondition.objects.filter(condition_group=data_condition_group)
-        if "threshold_type" in updated_fields:
-            threshold_type = (
-                Condition.GREATER
-                if updated_fields["threshold_type"] == AlertRuleThresholdType.ABOVE.value
-                else Condition.LESS
-            )
-            resolve_threshold_type = (
-                Condition.LESS_OR_EQUAL
-                if updated_fields["threshold_type"] == AlertRuleThresholdType.ABOVE.value
-                else Condition.GREATER_OR_EQUAL
-            )
-            for dc in data_conditions:
-                if dc.condition_result == DetectorPriorityLevel.OK:
-                    dc.update(type=resolve_threshold_type)
-                else:
-                    dc.update(type=threshold_type)
-
-        if "resolve_threshold" in updated_fields:
-            resolve_condition = data_conditions.get(condition_result=DetectorPriorityLevel.OK)
-            if updated_fields["resolve_threshold"] is None:
-                # we need to figure out the resolve threshold ourselves
-                resolve_threshold = get_resolve_threshold(data_condition_group)
-                if resolve_threshold != -1:
-                    resolve_condition.update(comparison=resolve_threshold)
-                else:
-                    raise UnresolvableResolveThreshold
-            else:
-                resolve_condition.update(comparison=updated_fields["resolve_threshold"])
-
-    detector.update(**updated_detector_fields)
+    update_detector(alert_rule, detector)
+
+    data_condition_group = detector.workflow_condition_group
+    if data_condition_group is None:
+        # this shouldn't be possible due to the way we dual write
+        logger.error(
+            "AlertRuleDetector has no associated DataConditionGroup",
+            extra={"alert_rule_id": alert_rule.id},
+        )
+        raise MissingDataConditionGroup
+    data_conditions = DataCondition.objects.filter(condition_group=data_condition_group)
+
+    # update the data condition types if the threshold type was updated
+
+    threshold_type = (
+        Condition.GREATER
+        if alert_rule.threshold_type == AlertRuleThresholdType.ABOVE.value
+        else Condition.LESS
+    )
+    resolve_threshold_type = (
+        Condition.LESS_OR_EQUAL
+        if alert_rule.threshold_type == AlertRuleThresholdType.ABOVE.value
+        else Condition.GREATER_OR_EQUAL
+    )
+    for dc in data_conditions:
+        if dc.condition_result == DetectorPriorityLevel.OK:
+            dc.update(type=resolve_threshold_type)
+        else:
+            dc.update(type=threshold_type)
```

Comment:
```suggestion
    for data_condition in data_conditions:
        if data_condition.condition_result == DetectorPriorityLevel.OK:
            data_condition.update(type=resolve_threshold_type)
        else:
            data_condition.update(type=threshold_type)
```
just a nit, i think it's a little easier to read the code with the full names.